### PR TITLE
Documentation for Out Of Memory Problems [DOC-386]

### PR DESCRIPTION
**Pull request by alexandernutz**
Jira ticket: [DOC-386](https://certora.atlassian.net/browse/DOC-386)
Link to generated documentation: https://certora-certora-prover-documentation--238.com.readthedocs.build/en/238/docs/user-guide/out-of-resources/index.html

[DOC-386]: https://certora.atlassian.net/browse/DOC-386?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/user-guide/out-of-resources/memout.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+places in the documentation on timeout prevention.
+
+(memout-indicators)=
+## General indicators
+
+When the free memory drops below a certain threshold, the Prover issues the
+following warning to the "General Problems" panel: `Extremely low available
+memory`.
+
+This warning might occasionally be a false positive: the JVM is sometimes able
```
[the remark this answers is not in the document]
Comment (alexandernutz) [reply docs/user-guide/out-of-resources/memout.md:16]: right right ..  may have cut too much / not reused something .. will look (or just go with your suggestion ..)

[the remark this answers is not in the document]
Comment (nafur) [reply docs/user-guide/out-of-resources/memout.md:16]: Actually I think this I already phrased it that way :-)

[the remark this answers is not in the document]
Comment (alexandernutz) [reply docs/user-guide/out-of-resources/memout.md]: .. and I broke it? 😳 
Alright, bringing the "not always be conclusive" back ...

File: docs/user-guide/out-of-resources/index.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ran out of time or out of memory.
+
+Out-of-memory problems are signified by an `Extremely low available memory`
+message in the Global Problems tab of the Prover reports, see
+{ref}`memout-introduction` for more details. Timeouts are signified either by a
+`Global timeout reached` message in the Global Problems tab, if the whole Prover
+job timed out, or by an orange clock symbol next to the rule, if that
+particular rule timed out, see {ref}`timeouts-introduction` for more details.
+
+% Unfortunately, the Certora Prover regularly encounters issues it can not resolve
```
[the remark this answers is not in the document]
Comment (alexandernutz) [reply docs/user-guide/out-of-resources/index.md]: oh, sorry, I did that later

File: docs/user-guide/out-of-resources/memout.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+to clean up enough memory on-demand to avert any crashes, or the memory might be
+just enough. It might also not show up although the job fails due to
+insufficient memory, e.g., if a single allocation that is greater than the
+warning threshold fails. Both cases are expected to be rare.
+
+% The prover log oftentimes contains other warnings that point to an out-of-memory
+% issue.
+
+
+% ### SMT solvers dying
```
[the remark this answers is not in the document]
Comment (alexandernutz) [reply docs/user-guide/out-of-resources/memout.md]: ditto -- I'll try to adapt to your pace better next time ;-)

**Pull request merged**